File: android_native/app/src/main/java/com/ezbillify/ezworkspace/ui/screens/DashboardShell.kt
package com.ezbillify.ezworkspace.ui.screens

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.BarChart
import androidx.compose.material.icons.filled.GridView
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Widgets
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.ui.platform.LocalContext
import com.ezbillify.ezworkspace.services.AuthService
import com.ezbillify.ezworkspace.services.AppUser
import com.ezbillify.ezworkspace.services.Prefs
import com.ezbillify.ezworkspace.ui.components.DraggableAiOrb
import com.ezbillify.ezworkspace.ui.components.EzNavBar
import com.ezbillify.ezworkspace.ui.components.NavDest
import com.ezbillify.ezworkspace.ui.components.rememberContainerSize
import kotlinx.coroutines.launch

/** App shell — bottom nav (Home/Modules/Activity/Profile) + a free-floating, draggable EZ AI orb. */
@Composable
fun DashboardShell(
    onOpenModule: (section: Int, item: Int) -> Unit,
    onOpenEzAi: () -> Unit,
    onLoggedOut: () -> Unit,
) {
    val context = LocalContext.current
    val auth = remember { AuthService() }
    val scope = rememberCoroutineScope()
    var index by remember { mutableIntStateOf(0) }
    var user by remember { mutableStateOf<AppUser?>(null) }

    LaunchedEffect(Unit) { user = auth.loadProfile() }

    fun logout() {
        scope.launch {
            Prefs(context).clearOrb()
            auth.signOut()
            onLoggedOut()
        }
    }

    val navItems = listOf(
        NavDest(Icons.Filled.GridView, "Home"),
        NavDest(Icons.Filled.Widgets, "Modules"),
        NavDest(Icons.Filled.BarChart, "Activity"),
        NavDest(Icons.Filled.Person, "Profile"),
    )

    val (containerSize, sizeModifier) = rememberContainerSize()

    Box(Modifier.fillMaxSize().then(sizeModifier)) {
        Box(Modifier.fillMaxSize().padding(bottom = 84.dp)) {
            when (index) {
                0 -> DashboardScreen(user = user, onOpenModule = onOpenModule)
                1 -> ModulesScreen(onOpenModule = onOpenModule)
                2 -> ActivityScreen()
                3 -> ProfileScreen(user = user, onLogout = ::logout)
            }
        }
        Box(Modifier.fillMaxSize().padding(horizontal = 16.dp).padding(bottom = 12.dp), contentAlignment = Alignment.BottomCenter) {
            EzNavBar(index = index, items = navItems, onTap = { index = it })
        }
        DraggableAiOrb(containerSize = containerSize, onTap = onOpenEzAi)
    }
}

File: flutter_app/lib/screens/dashboard_shell.dart
import 'package:flutter/material.dart';
import '../widgets/glass.dart';
import '../widgets/liquid_nav.dart';
import '../widgets/draggable_ai_orb.dart';
import '../services/auth_service.dart';
import '../services/prefs.dart';
import 'dashboard_screen.dart';
import 'modules_screen.dart';
import 'ez_ai_screen.dart';
import 'activity_screen.dart';
import 'profile_screen.dart';
import 'login_screen.dart';

/// App shell — Liquid Glass floating nav + a free-floating, draggable EZ AI orb.
class DashboardShell extends StatefulWidget {
  const DashboardShell({super.key});

  @override
  State<DashboardShell> createState() => _DashboardShellState();
}

class _DashboardShellState extends State<DashboardShell> {
  int _index = 0;
  AppUser? _user;
  final _auth = AuthService();

  static const _nav = [
    NavDest(Icons.grid_view_rounded, 'Home'),
    NavDest(Icons.widgets_outlined, 'Modules'),
    NavDest(Icons.bar_chart_rounded, 'Activity'),
    NavDest(Icons.person_outline, 'Profile'),
  ];

  @override
  void initState() {
    super.initState();
    _auth.loadProfile().then((u) {
      if (mounted) setState(() => _user = u);
    });
  }

  Future<void> _logout() async {
    await Prefs.clearOrb(); // re-center the orb on next login
    await _auth.signOut();
    if (!mounted) return;
    Navigator.of(context).pushAndRemoveUntil(
      MaterialPageRoute(builder: (_) => const LoginScreen()),
      (_) => false,
    );
  }

  void _openEzAi() {
    Navigator.of(context).push(MaterialPageRoute(
      builder: (_) => const Scaffold(
        body: GlassBackground(child: SafeArea(child: EzAiScreen())),
      ),
    ));
  }

  @override
  Widget build(BuildContext context) {
    final pages = [
      DashboardScreen(user: _user),
      const ModulesScreen(),
      const ActivityScreen(),
      ProfileScreen(user: _user, onLogout: _logout),
    ];

    return Scaffold(
      extendBody: true,
      body: Stack(
        children: [
          GlassBackground(
            child: SafeArea(bottom: false, child: IndexedStack(index: _index, children: pages)),
          ),
          // Liquid Glass floating nav
          Positioned(
            left: 0, right: 0, bottom: 0,
            child: SafeArea(
              top: false,
              child: LiquidNav(
                index: _index,
                items: _nav,
                onTap: (i) => setState(() => _index = i),
              ),
            ),
          ),
          // Free-floating, draggable EZ AI ball (tap → EZ AI, long-press → drag)
          DraggableAiOrb(onTap: _openEzAi),
        ],
      ),
    );
  }
}
